ray-operator/controllers/ray/common/pod.go

```diff
@@ -572,6 +572,19 @@ func setContainerEnvVars(pod *v1.Pod, rayContainerIndex int, rayNodeType rayiov1
 		portEnv := v1.EnvVar{Name: RAY_PORT, Value: headPort}
 		container.Env = append(container.Env, portEnv)
 	}
+
+	if !envVarExists(RAY_CLUSTER_NAME, container.Env) {
```

Comment:
In my opinion, this environment variable should always be maintained by KubeRay. Users should not be able to overwrite it.